gpflow/utilities/utilities.py

```diff
@@ -190,17 +190,22 @@ def leaf_components(input: tf.Module):
 def _merge_leaf_components(
     input: Dict[str, Union[tf.Variable, tf.Tensor, Parameter]]
 ) -> Dict[str, Union[tf.Variable, tf.Tensor, Parameter]]:
-    input_values = set([value.experimental_ref() for value in input.values()])
+
+    ref_fn = lambda x: (x if isinstance(x, Parameter) else x.ref())
```

Comment:
why not add a `ref` / `deref` method to `Parameter`? Seems like this is just missing from TFP's `TransformedVariable` API.
```
def ref(self):
    return self.unconstrained.ref()
```

Reply:
The parameter is not a tensor object and not a trait to the tensor class, it simply a wrapper to a tensor. You cannot make same reference for two different objects. That was one of the reasons why saving didn't work.

Reply:
Would it not be cleaner to have
```python
def ref(self):
    return self
```
 in order to get rid of `ref_fn` and `deref_fn`? It seems to me that this referencing functionality could be handy elsewhere as well at some point.

Reply:
The `ref()`  belongs to tensor API, and `ref()` returns a different type. There must be a distinction between these two for clarity. I would prefer not to introduce this method to the parameter class because of a printing issue.